Examples/Algorithms/TrackFinding/src/TrackParamsEstimationAlgorithm.cpp

```diff
@@ -0,0 +1,113 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#include "ActsExamples/TrackFinding/TrackParamsEstimationAlgorithm.hpp"
+
+#include "Acts/EventData/TrackParameters.hpp"
+#include "Acts/Seeding/EstimateTrackParamsFromSeed.hpp"
+#include "ActsExamples/EventData/IndexSourceLink.hpp"
+#include "ActsExamples/EventData/SimSeed.hpp"
+#include "ActsExamples/EventData/Track.hpp"
+#include "ActsExamples/Framework/WhiteBoard.hpp"
+
+#include <map>
+#include <stdexcept>
+
+ActsExamples::TrackParamsEstimationAlgorithm::TrackParamsEstimationAlgorithm(
+    ActsExamples::TrackParamsEstimationAlgorithm::Config cfg,
+    Acts::Logging::Level lvl)
+    : ActsExamples::BareAlgorithm("TrackParamsEstimationAlgorithm", lvl),
+      m_cfg(std::move(cfg)) {
+  if (m_cfg.inputSeeds.empty()) {
+    throw std::invalid_argument("Missing input seeds collection");
+  }
+  if (m_cfg.inputSourceLinks.empty()) {
+    throw std::invalid_argument("Missing input source links collection");
+  }
+  if (m_cfg.outputTrackParameters.empty()) {
+    throw std::invalid_argument("Missing output track parameters collection");
+  }
+  if (m_cfg.outputTrackParametersSeedMap.empty()) {
+    throw std::invalid_argument(
+        "Missing output trackparameters-to-seed collection");
+  }
+  if (not m_cfg.trackingGeometry) {
+    throw std::invalid_argument("Missing tracking geometry");
+  }
+
+  // Set up the track parameters covariance (the same for all tracks)
+  m_cfg.covariance(Acts::eBoundLoc0, Acts::eBoundLoc0) =
+      m_cfg.sigmaLoc0 * m_cfg.sigmaLoc0;
+  m_cfg.covariance(Acts::eBoundLoc1, Acts::eBoundLoc1) =
+      m_cfg.sigmaLoc1 * m_cfg.sigmaLoc1;
+  m_cfg.covariance(Acts::eBoundPhi, Acts::eBoundPhi) =
+      m_cfg.sigmaPhi * m_cfg.sigmaPhi;
+  m_cfg.covariance(Acts::eBoundTheta, Acts::eBoundTheta) =
+      m_cfg.sigmaTheta * m_cfg.sigmaTheta;
+  m_cfg.covariance(Acts::eBoundQOverP, Acts::eBoundQOverP) =
+      m_cfg.sigmaQOverP * m_cfg.sigmaQOverP;
+  m_cfg.covariance(Acts::eBoundTime, Acts::eBoundTime) =
+      m_cfg.sigmaT0 * m_cfg.sigmaT0;
```

Comment:
The covariance should be a private member of the algorithm. It is automatically defined by the configuration and any setting by the user is overridden anyway.

Reply:
Fixed.